test_helpers/main.go

```diff
@@ -0,0 +1,238 @@
+	// https://www.tarantool.io/en/doc/latest/reference/configuration/#cfg-basic-listen
+	Listen string
+
+	// WorkDir is box.cfg work_dir parameter for tarantool.
+	// Specify folder to store tarantool data files.
+	// Folder must be unique for each tarantool process used simultaneously.
+	// https://www.tarantool.io/en/doc/latest/reference/configuration/#confval-work_dir
+	WorkDir string
+
+	// User is a username used to connect to tarantool.
+	// All required grants must be given in InitScript.
+	User string
+
+	// Pass is a password for specified User.
+	Pass string
+
+	// WaitStart is a time to wait before starting to ping tarantool.
+	WaitStart time.Duration
+
+	// ConnectRetry is a count of attempts to ping tarantool.
+	ConnectRetry uint
+
+	// RetryTimeout is a time between tarantool ping retries.
+	RetryTimeout time.Duration
+}
+
+// TarantoolInstance is a data for instance graceful shutdown and cleanup.
+type TarantoolInstance struct {
+	// Cmd is a Tarantool command. Used to kill Tarantool process.
+	Cmd *exec.Cmd
+
+	// WorkDir is a directory with tarantool data. Cleaned up after run.
+	WorkDir string
+}
+
+func isReady(server string, opts *tarantool.Opts) error {
+	var err error
+	var conn *tarantool.Connection
+	var resp *tarantool.Response
+
+	conn, err = tarantool.Connect(server, *opts)
+	if err != nil {
+		return err
+	}
+	if conn == nil {
+		return errors.New("Conn is nil after connect")
+	}
+	defer conn.Close()
+
+	resp, err = conn.Ping()
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return errors.New("Response is nil after ping")
+	}
+
+	return nil
+}
+
+var (
+	// Used to extract Tarantool version (major.minor.patch).
+	tarantoolVersionRegexp *regexp.Regexp
+)
+
+func init() {
+	tarantoolVersionRegexp = regexp.MustCompile(`Tarantool (?:Enterprise )?(\d+)\.(\d+)\.(\d+).*`)
+}
+
+// atoiUint64 parses string to uint64.
+func atoiUint64(str string) (uint64, error) {
+	res, err := strconv.ParseUint(str, 10, 32)
+	if err != nil {
+		return 0, fmt.Errorf("cast to number error (%s)", err)
+	}
+	return res, nil
+}
+
+// IsTarantoolVersionLess checks if tarantool version is less
+// than passed <major.minor.patch>. Returns error if failed
+// to extract version.
+func IsTarantoolVersionLess(majorMin uint64, minorMin uint64, patchMin uint64) (bool, error) {
+	var major, minor, patch uint64
+
+	out, err := exec.Command("tarantool", "--version").Output()
+
+	if err != nil {
+		return true, err
+	}
+
+	parsed := tarantoolVersionRegexp.FindStringSubmatch(string(out))
+
+	if parsed == nil {
+		return true, errors.New("regexp parse failed")
+	}
+
+	if major, err = atoiUint64(parsed[1]); err != nil {
+		return true, fmt.Errorf("failed to parse major: %s", err)
+	}
+
+	if minor, err = atoiUint64(parsed[2]); err != nil {
+		return true, fmt.Errorf("failed to parse minor: %s", err)
+	}
+
+	if patch, err = atoiUint64(parsed[3]); err != nil {
+		return true, fmt.Errorf("failed to parse patch: %s", err)
+	}
+
+	if major < majorMin {
```

Comment:
```
if major != majorMin {
    return major < majorMin, nil
} else if minor != minorMin {
   return minor < minorMin, nil
} else {
    return patch < patchMin, nil
}
```

Reply:
That's shorter and not really harder to understand. Thank you!